include/BitFieldTools.h

```diff
@@ -46,27 +46,38 @@
     #include "DataTypes.h"
 #endif
 
-namespace Mezzanine {
+namespace Mezzanine
+{
+    /// @brief Convenience method for getting the value of the n-th bit.
+    /// @remarks This is useful for setting values on bit-fields.  A value of 0 returns 0,
+    /// a value of 1 returns 1, 2 returns 2, 3 returns 4, 4 returns 8, 5 returns 16, etc..
+    /// @param Bit The n-th bit to get the value of.  Range of 0-64.
+    /// @return Returns the value at the specified bit position.
+    constexpr UInt64 EnumBit(const UInt32 Bit)
+    {
+        UInt64 BitValue = 1;
+        return ( Bit == 0 ? 0 : BitValue << ( Bit - 1 ) );
```

Comment:
Undefined behavior here when Bit is 0. Because ( Bit - 1 ) == UINT_MAX and 1 << UINT_MAX is UB.

Reply:
So now these show up in the main comment feed... I guess I will use these.

Reply:
As stated in your edit above, I do have a check for zero.  I don't have a check for >64 though, and could add that.